### ISSUE TITLE
Handle "Queued to Merge" status

### ISSUE THREAD

**Issue by pcorpet** [Reviewable/Reviewable#1053]
As of 2023-05-09, Reviewable ignores the "queued to merge" status.

Merge queues are a way to get PR merged that is a bit new to GitHub
https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/managing-a-merge-queue

Having a nice understanding in Reviewable that this is getting merged (i.e. no need of attention) would be useful.

**Comment by pkaminski** [comment]
Good idea.  What information beyond "is queued" would you like to see?  The API tells us who enqueued it, when, on which queue, what the status is (awaiting checks, locked, queued, etc...), when it's expected to get merged, etc.

We won't be able to get to this right away, as 1) we'll need a place to display this information (likely a new panel dedicated to PR information), and 2) we'll need to switch to GraphQL for our PR data fetch as the queuing info is not available via the REST API.

**Comment by pcorpet** [comment]
the main one for us is "queued", GitHub differentiates it very early in the UI (color + icon). It's important to distinguish from Draft/Open/**Queued**/Merged/Closed. For now in Reviewable they appear as "Open ready to Merge" so I have those big green buttons to squash and merge that look so tempting but should not be used.

After that the additional info (which queue, time to get merged, etc.) is just bonus.

**Comment by pkaminski** [comment]
BTW, does Reviewable's current merge button work for queuing a PR for merge?  I'm guessing not, in which case we should also add support for queuing / dequeuing a PR as part of this fix.

**Comment by pcorpet** [comment]
No, Reviewable button actually merges things directly for the PR.

**Comment by pkaminski** [comment]
Another idea mentioned in #1099 is to ensure that queued PRs are shifted out of any for-review sections on the dashboard -- perhaps they should get their own section?
